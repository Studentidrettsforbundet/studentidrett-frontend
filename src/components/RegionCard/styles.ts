import { style, classes } from 'typestyle';
import colors from '../../styles/colors';

export const regionCard = classes(
    style({
        cursor: 'pointer',
        margin: '0 auto 16px',
        padding: '10px',
        maxWidth: '500px',
        backgroundColor: colors.cardHeader,
        filter: 'drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25))',
        borderRadius: '5px',
        $nest: {
            '& div > h5': {
                color: colors.secondary,
                textAlign: 'center',
                marginTop: 'auto',
                marginBottom: 'auto',
            },
        },
    }),
);
