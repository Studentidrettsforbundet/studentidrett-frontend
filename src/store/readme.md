# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
