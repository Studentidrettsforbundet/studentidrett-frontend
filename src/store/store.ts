import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
