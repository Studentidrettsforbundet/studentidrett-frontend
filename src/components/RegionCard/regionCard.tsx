import React, { useState } from 'react';
import { regionInterface } from '../../interfaces';
import CityCard from '../CityCard/cityCard';
import { cardBody, cardTitle,  } from '../../styles/card';
import {regionCard} from './styles'

// See: https://getbootstrap.com/docs/4.0/components/card/

const RegionCard = ({ id, name, cities }: regionInterface) => {
    const [expanded, setExpanded] = useState(false);

    const toggleExpanded = () => {
        setExpanded(!expanded);
    };

    return (

        <div className={regionCard} key={id} onClick={() => toggleExpanded()}>
            <div>
                <h5 className="boldText">{name}</h5>
                {expanded ? (
                    cities.map((entry) => {
                        return (
                            <CityCard
                                id={entry.id} 
                                name={entry.name}
                                region={entry.region}
                                clubs={entry.clubs}
                            />
                        );
                    })
                ) : (<div></div>
                )}
            </div>
        </div>
    );
};

export default RegionCard;
