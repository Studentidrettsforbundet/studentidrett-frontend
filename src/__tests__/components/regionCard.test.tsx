import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import RegionCard from '../../components/regionCard';
import store from '../../store/store';


describe('regionCard', () => {
    test("renders correctly", () => {
        const tree = renderer
            .create(<Provider store={store}>
                <MemoryRouter>
                    <RegionCard id={0} name={"MidtNorge"} cities={[]}/>
                </MemoryRouter>
            </Provider>);
        expect(tree).toMatchSnapshot();
    });
});