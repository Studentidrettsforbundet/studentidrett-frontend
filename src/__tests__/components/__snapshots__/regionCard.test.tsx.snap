// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regionCard renders correctly 1`] = `
<div
  className="f1ovq3do fq2301v"
  onClick={[Function]}
>
  <div>
    <h5>
      Vestlandet
    </h5>
    <div />
  </div>
</div>
`;
